jdi-light/src/main/java/com/epam/jdi/light/elements/base/JDIBase2.java

```diff
@@ -0,0 +1,525 @@
+
+    public JDILocator locator = new JDILocator(this);
+    @Override
+    public DriverBase setParent(Object parent) {
+        //this.locator.isRoot = false;
+        return super.setParent(parent);
+    }
+    public CacheValue<WebElement> webElement = new CacheValue<>();
+    public CacheValue<List<WebElement>> webElements = new CacheValue<>();
+    public MapArray<String, JFunc1<WebElement, Boolean>> searchRules = new MapArray<>();
+    private MapArray<String, JFunc1<WebElement, Boolean>> searchRules() {
+        return searchRules;
+    }
+    public JAction1<UIElement> beforeSearch = null;
+    WebElement beforeSearch(WebElement el) {
+        (beforeSearch == null ? ELEMENT.beforeSearch : beforeSearch).execute(new UIElement(el));
+        return el;
+    }
+
+    public JDIBase doBefore(JAction1<UIElement> action) { beforeSearch = action; return this; }
+    public JDIBase showBefore() { beforeSearch = UIElement::show; return this; }
+    public JDIBase noValidation() {
+        return setSearchRule("Any", ANY_ELEMENT);
+    }
+    public <T> T noValidation(JFunc<T> func) {
+        MapArray<String, JFunc1<WebElement, Boolean>> rules = searchRules.copy();
+        searchRules.clear();
+        T result = func.execute();
+        searchRules = rules;
+        return result;
+    }
+    public JDIBase searchVisible() {
+        return setSearchRule("Visible", VISIBLE_ELEMENT);
+    }
+    public JDIBase visibleEnabled() { return setSearchRule("Enabled", ENABLED_ELEMENT); }
+    public JDIBase inView() {
+        showBefore();
+        return setSearchRule("Element in view", ELEMENT_IN_VIEW);
+    }
+    public JDIBase addSearchRule(String name, JFunc1<WebElement, Boolean> rule) {
+        searchRules.add(name, rule);
+        return this;
+    }
+    public JDIBase setSearchRule(String name, JFunc1<WebElement, Boolean> rule) {
+        searchRules.clear();
+        searchRules.add(name, rule);
+        return this;
+    }
+    public JDIBase setWebElement(WebElement el) {
+        webElement.setForce(el);
+        return this;
+    }
+    public JDIBase setWebElements(List<WebElement> els) {
+        webElements.setForce(els);
+        return this;
+    }
+
+    public JDIBase setLocator(String locator) {
+        return setLocator(defineLocator(locator));
+    }
+    public JDIBase setLocator(By locator) {
+        if (locator != null) {
+            if (name.isEmpty())
+                name = shortBy(locator, this);
+            this.locator.add(locator, this);
+        }
+        return this;
+    }
+    public JDIBase setFrames(List<By> frames) {
+        if (name.isEmpty() && frames.size() > 0)
+            name = shortBy(frames.get(0));
+        this.locator.add(frames, this);
+        return this;
+    }
+    public By getLocator(Object... args) {
+        initContext();
+        return locator.getLocator(args);
+    }
+    public List<By> getFrames() { return locator.getFrames(); }
+
+    protected int timeout = -1;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5327794)